internal/status/gatewayclassconditions.go

```diff
@@ -61,7 +61,10 @@ func mergeConditions(conditions []metav1.Condition, updates ...metav1.Condition)
 					conditions[j].Reason = update.Reason
 					conditions[j].Message = update.Message
 					conditions[j].ObservedGeneration = update.ObservedGeneration
-					conditions[j].LastTransitionTime = update.LastTransitionTime
+					// Only update the transition time if Status changes.
+					if cond.Status != update.Status {
```

Comment:
Makes sense and consistent with the godoc for `LastTransitionTime`, which says "lastTransitionTime is the last time the condition transitioned from one status to another."

Also just noting for reference that https://github.com/projectcontour/contour/blob/main/internal/k8s/helpers.go#L29 is used when determining whether to actually write status updates to the apiserver, to filter out transition time-only changes. But it's an improvement to not have changes like that in the first place.